operator/provider_azure_flags.go

```diff
@@ -26,9 +26,6 @@ import (
 func init() {
 	flags := rootCmd.Flags()
 
-	flags.String(operatorOption.AzureCloudName, "AzurePublicCloud", "Name of the Azure cloud being used")
-	option.BindEnvWithLegacyEnvFallback(operatorOption.AzureCloudName, "AZURE_CLOUD_NAME")
-
```

Comment:
This changed was introduced in 1.9 so we shouldn't simply remove it.

One way to fix it is to mart the option as deprecated with `flags.MarkDeprecated()` and add it as part of the upgrade notes for 1.11 and simply state that this will be removed since it's auto-detected now.

Reply:
Alright, thank you!

Reply:
Talked with @aanm on Slack and got a pointer on using `viper.IsSet` for checking if the cloud name CLI flag is passed in order to trigger the new behavior. This should allow the current users which override it to keep the behavior that they wanted and the users who do not override the flag to get the new auto-detection. Let me know if you folks agree with this.